### PR TITLE
Add generic types to the client registry for registering TESR's

### PR DESCRIPTION
**Pull request by AlexIIL**
This also removes the unchecked cast warning. I assume this will be compatible with all current mods, as it caused no compilation errors in my workspace.

### CODE REVIEW

**Comment by cpw** [comment]
Yeah we have dozens of generic cleanups like this..

**Comment by AlexIIL** [comment]
Do you want to do them yourself or would you appreciate help? (And then squashing all of the changes into a single PR)

**Comment by cpw** [comment]
I welcome PRs where others go through. Avoid the registry, its getting a complete overhaul, its half done here ATM..

**Comment by AlexIIL** [comment]
Should I avoid all Registries? Or just this and the GameRegistry?

File: src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java [hunk 1]
```diff
@@ -3,10 +3,8 @@
 import net.minecraft.block.Block;
 import net.minecraft.client.renderer.EntityRenderer;
 import net.minecraft.entity.Entity;
-import net.minecraft.entity.EntityLivingBase;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
-import net.minecraftforge.fml.common.eventhandler.Event.Result;
```
Comment (AlexIIL) [inline src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java]: I didn't originally intend to do this, and this is the only import change I have done.

File: src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public static FakePlayer getMinecraft(WorldServer world)
         if (ret == null)
         {
             ret = FakePlayerFactory.get(world,  MINECRAFT);
-            MINECRAFT_PLAYER = new WeakReference(ret);
+            MINECRAFT_PLAYER = new WeakReference<FakePlayer>(ret);
```
Comment (AlexIIL) [inline src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java]: This must have been done on purpose, but for simplicities sake I have changed it as well (Unless there was a specific reason to NOT have a generic type here that I am unaware of)

File: src/main/java/net/minecraftforge/fluids/BlockFluidBase.java [hunk 3]
```diff
@@ -90,11 +90,11 @@
     public static final PropertyInteger LEVEL = PropertyInteger.create("level", 0, 15);
     public static final PropertyFloat[] LEVEL_CORNERS = new PropertyFloat[4];
     public static final PropertyFloat FLOW_DIRECTION = new PropertyFloat("flow_direction");
-    public static final IUnlistedProperty[] FLUID_RENDER_PROPS;
+    public static final IUnlistedProperty<Float>[] FLUID_RENDER_PROPS;
 
     static
     {
-        ImmutableList.Builder<IUnlistedProperty> builder = ImmutableList.builder();
+        ImmutableList.Builder<IUnlistedProperty<Float>> builder = ImmutableList.builder();
```
Comment (AlexIIL) [inline src/main/java/net/minecraftforge/fluids/BlockFluidBase.java]: This is potentially a breaking change for existing mods, however I assume no mod will have a problem with being told that they are all float values.

File: src/main/java/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java [hunk 4]
```diff
@@ -257,7 +257,7 @@ else if (state != ConnectionState.CONNECTED && state != ConnectionState.HANDSHAK
         }
     }
 
-    private boolean handleVanilla(Packet msg)
+    private boolean handleVanilla(Packet<?> msg)
```
Comment (AlexIIL) [inline src/main/java/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java]: I'm not sure you want this, as I haven't figured out how the whole class interacts with everything to generic type it properly, so I have left most network classes untouched.

File: src/main/java/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java [hunk 5]
```diff
@@ -160,7 +160,7 @@ public SimpleNetworkWrapper(String channelName)
 
     /**
      * Construct a minecraft packet from the supplied message. Can be used where minecraft packets are required, such as
-     * {@link TileEntity#func_145844_m}.
+     * {@link TileEntity#getDescriptionPacket}.
```
Comment (AlexIIL) [inline src/main/java/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java]: Might help for understanding :)

**Comment by AlexIIL** [comment]
I will squash them down when/if you are happy with the changes.

File: src/main/java/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java (hunk 5 above)
Comment (LexManos) [inline src/main/java/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java]: No, our comments should be in srg name as they get remapped in the end modder's setup to whatever mapping they are using.

Comment (AlexIIL) [inline src/main/java/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java]: Thanks for pointing that out. Fixed.

**Comment by AlexIIL** [comment]
Closing as Lex fixed this in MinecraftForge/MinecraftForge@5549b06f966d6129cd5b82e8328cea57a5c88c9e. If I find any more though I will re-open a PR

**Pull request closed without merging**